knotx-core/knotx-server/src/main/java/com/cognifide/knotx/server/KnotxServerVerticle.java

```diff
@@ -71,34 +71,34 @@ public void start(Future<Void> fut) throws IOException, URISyntaxException {
                 request -> {
                     request.setExpectMultipart(true);
                     request.endHandler(aVoid -> eventBus.<JsonObject>sendObservable(repositoryAddress, createRepositoryRequest(request))
-                        .doOnNext(this::traceMessage)
-                        .subscribe(
-                                reply -> {
-                                    RepositoryResponse repository = new RepositoryResponse(reply.body());
-                                    if (repository.isSuccess()) {
-                                        eventBus.<JsonObject>sendObservable(engineAddress, createEngineRequest(repository, request))
-                                                .subscribe(
-                                                        result -> {
-                                                            TemplateEngineResponse engineResponse = new TemplateEngineResponse(result.body());
-                                                            if (engineResponse.isSuccess()) {
-                                                                rewriteHeaders(request, request.headers());
-                                                                request.response().end(engineResponse.getHtml());
-                                                            } else {
-                                                                request.response().setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code()).end(engineResponse.getReason());
+                            .doOnNext(this::traceMessage)
+                            .subscribe(
+                                    reply -> {
+                                        RepositoryResponse repository = RepositoryResponse.fromJson(reply.body());
+                                        if (repository.shouldProcess()) {
+                                            eventBus.<JsonObject>sendObservable(engineAddress, createEngineRequest(repository, request))
+                                                    .subscribe(
+                                                            result -> {
+                                                                TemplateEngineResponse engineResponse = new TemplateEngineResponse(result.body());
+                                                                if (engineResponse.isSuccess()) {
+                                                                    rewriteHeaders(request, request.headers());
+                                                                    request.response().end(engineResponse.getHtml());
+                                                                } else {
+                                                                    request.response().setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code()).end(engineResponse.getReason());
```

Comment:
In one place we use plain integers as status codes (e.g. `RepositoryResponse`). Here those statuses are taken from `HttpResponseStatus`. Please unify status codes usage (preferably `HttpResponseStatus` ).